packages/@o3r/configuration/src/services/configuration/configuration.base.service.ts

```diff
@@ -116,32 +120,19 @@ export class ConfigurationBaseService {
    * @param id Id of the component
    */
   public getConfig(id: string): Observable<any> {
+    const globalConfig$ = this.store.pipe(select(selectGlobalConfig));
     return combineLatest([
-      this.store.pipe(
-        select(selectConfigurationEntities),
-        map((storedConfigs) => {
-          const globalConfigId = 'global';
-          const componentConfig = storedConfigs[id];
-          const globalConfig = storedConfigs[globalConfigId];
-          if (id !== globalConfigId && globalConfig) {
-            return {
-              ...globalConfig,
-              ...(componentConfig || {})
-            };
-          }
-          return componentConfig || null;
-        }),
-        distinctUntilChanged((prev, current) => JSON.stringify(prev) === JSON.stringify(current))
-      ), this.store.pipe(
-        select(selectComponentOverrideConfig(id))
-      )]
-    ).pipe(
-      map(([storeConfig, storeOverrideConfig]) => {
-        return {
-          ...storeConfig,
-          ...storeOverrideConfig
-        };
-      })
+      id === globalConfigId ? globalConfig$ : combineLatest([
+        globalConfig$,
+        this.store.pipe(select(selectConfigurationForComponent({ id })))
+      ]).pipe(
+        map(([globalConfig, componentConfig]) => ({ ...globalConfig, ...componentConfig })),
+        distinctUntilChanged(jsonStringifyDiff)
+      ),
+      this.store.pipe(select(selectComponentOverrideConfig(id)))
+    ]).pipe(
+      map(([storeConfig, storeOverrideConfig]) => ({ ...storeConfig, ...storeOverrideConfig })),
+      distinctUntilChanged(jsonStringifyDiff)
```

Comment:
can you avoid the double `combineLatest` with something like :
```suggestion
     globalConfig$,
     id === globalConfigId ? of({}) : this.store.pipe(select(selectConfigurationForComponent({ id }))),
      this.store.pipe(select(selectComponentOverrideConfig(id)))
    ]).pipe(
      map(([globalConfig, componentConfig, storeOverrideConfig]) => ({...globalConfig, ...componentConfig, ...storeOverrideConfig })),
      distinctUntilChanged(jsonStringifyDiff)
```
?